app/src/main/java/org/sopt/sample/SignUpActivity.kt

```diff
@@ -0,0 +1,57 @@
+package org.sopt.sample
+
+import android.content.Intent
+import androidx.appcompat.app.AppCompatActivity
+import android.os.Bundle
+import com.google.android.material.snackbar.Snackbar
+import org.sopt.sample.databinding.ActivitySignUpBinding
+
+class SignUpActivity : AppCompatActivity() {
+    private lateinit var binding : ActivitySignUpBinding
+
+    // 회원가입 성공 조건
+    private fun isValidId(id : String) : Boolean {
+        return id.length in 6..10
+    }
+    private fun isValidPw(pw : String) : Boolean {
+        return pw.length in 8..12
+    }
+
+    // 성공 여부 판단
+    private fun checkSignup() {
+        val nameText = binding.signupNameInput.text.toString()
+        val idText = binding.signupIdInput.text.toString()
+        val pwText = binding.signupPwInput.text.toString()
+        val mbtiText = binding.signupMbtiInput.text.toString()
+
+        if (isValidId(idText) && isValidPw(pwText)) {
+            intentToLogin(nameText, idText, pwText, mbtiText)
+        } else if (isValidId(idText)) {
+            Snackbar.make(binding.root, "비밀번호를 8-12글자 사이로 설정하세요", Snackbar.LENGTH_SHORT).show()
+        } else {
+            Snackbar.make(binding.root, "아이디를 6-10글자 사이로 설정하세요", Snackbar.LENGTH_SHORT).show()
```

Comment:
```suggestion
        if (isValidId(idText) && isValidPw(pwText)) {
            intentToLogin(nameText, idText, pwText, mbtiText)
            return
        }
        if (isValidId(idText)) {
            Snackbar.make(binding.root, "비밀번호를 8-12글자 사이로 설정하세요", Snackbar.LENGTH_SHORT).show()
            return
        }
            Snackbar.make(binding.root, "아이디를 6-10글자 사이로 설정하세요", Snackbar.LENGTH_SHORT).show()
```

early return 코드로 바꿔보면 좋을 것 같아요!

Reply:
오 early return 사용하면 가독성이 더 좋아질 것 같네요...! 한번 써보겠습니다!